app/app.js

```diff
@@ -0,0 +1,61 @@
+import $ from "jquery";
+import * as tournament from "./tournament/tournament";
+import * as player from "./tournament/player";
+
+var mTournament = new tournament.Tournament();
+
+$(document).ready(function() {
+    $("#buttonAdd").click(addPlayerIfValid);
+    $("#buttonCreate").click(generateTournament);
+});
+
+function addPlayerIfValid() {
+    var info = getPlayerInfo();
+
+    try {
+        var newPlayer = new player.Player(info.name, info.rank);
+        mTournament.addPlayer(newPlayer);
+        updateTable(newPlayer);
+
+    } catch (e) {
+        window.alert("Not a valid player");
+    }
+}
+
+function getPlayerInfo() {
+    return {
+        name: $("#fieldName")[0].value,
+        rank: $("#fieldRank")[0].value
+    }
+}
+
+function updateTable(player) {
+    clearFields();
+    addPlayerToTable(player);
+}
+
+function clearFields() {
+    $("#fieldName")[0].value = "";
+    $("#fieldRank")[0].value = "";
+}
+
+function addPlayerToTable(player) {
+    $("#playerListBody").append(newRow(player));
+}
+
+var newRow = (player) => `<tr><td>${player.name.replace(/</g, "&lt;").replace(/>/g, "&gt;")}</td><td>${player.rank}</td></tr>`;
+
+function generateTournament() {
+    try {
+        mTournament.generate();
+        closeEnrollment();
+
+    } catch (e) {
+        window.alert("Too few players!");
```

Comment:
This is one of the reasons we should not use exceptions. Many other things can go wrong, and you´ll end up showing the wrong message.